src/js/components/PodSpecView.js

```diff
@@ -0,0 +1,218 @@
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import PodSpec from '../structs/PodSpec';
+import PodContainerSpecView from './PodContainerSpecView';
+import PodNetworkSpecView from './PodNetworkSpecView';
+
+const METHODS_TO_BIND = [
+];
+
+class PodSpecView extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getEnvironmentDetails() {
+    let {spec} = this.props;
+    let environment = Object.assign({}, spec.getEnvironment());
+
+    if (Object.keys(environment).length === 0) {
+      return null;
+    }
+
+    Object.keys(environment).forEach(function (key) {
+      let value = environment[key];
+      if (typeof value === 'object' && !Array.isArray(value) &&
+          value !== null) {
+        environment[key] = value.secret;
```

Comment:
Let's put a `Key Icon + Referred key name` instead of just the reference to the secret